src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+        actual3.setLogin("Snowhunter");
+        actual3.setPassword("123456");
+        actual3.setAge(41);
+        User expected = registrationService.register(actual);
+        assertEquals(expected, actual);
+        expected = registrationService.register(actual1);
+        assertEquals(expected, actual1);
+        expected = registrationService.register(actual2);
+        assertEquals(expected, actual2);
+        expected = registrationService.register(actual3);
+        assertEquals(expected, actual3);
+    }
+
+    @Test
+    void register_theSameLogin_NotOk() {
+        User testUser = new User();
+        testUser.setLogin("WormJim");
+        testUser.setPassword("123456");
+        testUser.setAge(29);
+        registrationService.register(testUser);
+        User testUser2 = new User();
+        testUser2.setLogin("WormJim");
+        testUser2.setPassword("654321");
+        testUser2.setAge(24);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser2);
+        });
+    }
+
+    @Test
+    void register_invalidLoginLength_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Jo");
+        testUser.setPassword("123456");
+        testUser.setAge(24);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setLogin("");
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setLogin("J");
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_invalidPasswordLength_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Snowman");
+        testUser.setPassword("123");
+        testUser.setAge(32);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setPassword("");
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setPassword("12345");
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Snowman");
+        testUser.setPassword("123456");
+        testUser.setAge(17);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setAge(-1);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setAge(0);
+        assertThrows(ValidationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
```

Comment:
This test method is checking multiple cases of invalid age. It's better to split it into several smaller test methods, each checking one specific case. This way, when a test fails, you'll know exactly what case is not working, just by looking at the test method's name.